src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1914,7 +1914,17 @@ private AssemblySymbol GetForwardedToAssembly(string fullName, int arity, out bo
 
         internal static void CheckFeatureAvailability(Location location, MessageID feature, DiagnosticBag diagnostics)
         {
-            LanguageVersion availableVersion = ((CSharpParseOptions)location.SourceTree.Options).LanguageVersion;
+            var options = (CSharpParseOptions)location.SourceTree.Options;
```

Comment:
I think @gafter designed this section so we should probably have him weigh in here on its intended usage strategy.

From what I can see the feature gating on syntax and compilation are separate, but interacting. In parsing I would expect only the parse flags on the parse options to be checked, but here, in the binder, I would expect both the parse options AND the compilation options to be checked in case the feature has either new syntax or new semantics.

I could also see the argument that only the compilation options need to be checked because presumably the parser would have gone through and added any parse-level feature diagnostics already.